### PR TITLE
eth/download/statesync : state hash sum optimized

### PR DESCRIPTION
**Pull request by ucwong**

### CODE REVIEW

**Comment by rjl493456442** [comment]
Is there any benefit? E.g. is there any performance speedup? Would be nice to have some benchmarks.

We are inclined to not accept the changes if it's basically noop.

File: eth/downloader/statesync.go [hunk 1]
```diff
@@ -18,17 +18,16 @@ package downloader
...
 	"github.com/ethereum/go-ethereum/common"
 	"github.com/ethereum/go-ethereum/core/rawdb"
 	"github.com/ethereum/go-ethereum/core/state"
+	"github.com/ethereum/go-ethereum/crypto"
 	"github.com/ethereum/go-ethereum/ethdb"
 	"github.com/ethereum/go-ethereum/log"
 	"github.com/ethereum/go-ethereum/trie"
 	"golang.org/x/crypto/sha3"
+	"sync"
+	"time"
```
Comment (holiman) [inline eth/downloader/statesync.go:30]: `goimports` needed

**Comment by holiman** [comment]
> Is there any benefit? E.g. is there any performance speedup? Would be nice to have some benchmarks.

Yes, `Read` avoid a copy which `Sum` does

**Comment by holiman** [comment]
Similar one in a different area: https://github.com/ethereum/go-ethereum/pull/22023/files

File: eth/downloader/statesync.go (hunk 1 above)
Comment (ucwong) [reply eth/downloader/statesync.go:30]: I have done the ```go fmt```, seems no change

**Review by holiman — APPROVED** [review]
LGTM

**Comment by holiman** [comment]
Thanks!

**Pull request merged**